packages/styled-components/src/sheet/GroupedTag.js

```diff
@@ -2,13 +2,14 @@
 /* eslint-disable no-use-before-define */
 
 import type { GroupedTag, Tag } from './types';
+import throwStyledError from '../utils/error';
 
 /** Create a GroupedTag with an underlying Tag implementation */
 export const makeGroupedTag = (tag: Tag): GroupedTag => {
   return new DefaultGroupedTag(tag);
 };
 
-const BASE_SIZE = 1 << 8;
+const BASE_SIZE = 1 << 9;
```

Comment:
I've increased the base-size here since I find it highlight likely that a lot of apps will actually have around 512 components usually. Larger apps will of course see this grow as usual